### PR TITLE
implement /proc parsing

### PR DESCRIPTION
**Pull request by vrothberg**
Parse all relevant files in /proc and /dev to fully replace
github.com/shirou/gopsutil.  Also add unit tests, which are now
possible as we can test our parsers.

Nothing has changed with regards to the API as containers/psgo is
currently only meant to replace ps (1) and not to offer a generic
library to parse /proc and /dev/{tty*,pts/*}.  From a technical
point of view, all code is in containers/psgo/pg to avoid exposing
any public API.

Performance-wise, we are around twice as fast as before as we
pre-load all data and do not load on-demand which is costly.
Furthermore, we now correctly treat tty devices and have a
consistent way of dealing with race conditions.

Note that the parses extract more data from /proc/$pid/stat and
from /proc/$pid/status than necessary.  This way, we can easily
extend the format descriptors to return more information.
Otherwise, the parsing of redundant fields can easily be removed.

Fixes: #4
Fixes: #7
Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@giuseppe @rhatdan PTAL

File: ps/ps.go [hunk 1]
```diff
@@ -12,13 +11,237 @@ import (
...
-	"github.com/shirou/gopsutil/process"
 	"golang.org/x/sys/unix"
 )
 
+// DefaultFormat is the `ps -ef` compatible default format.
+const DefaultFormat = "user,pid,ppid,pcpu,etime,tty,time,comm"
+
+var (
+	// ErrUnkownDescriptor is returned when an unknown descriptor is parsed.
+	ErrUnkownDescriptor = errors.New("unkown descriptor")
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply ps/ps.go]: Good catch! Thanks for checking.

**Comment by rhatdan** [comment]
LGTM
Could you add a field label to extract the SELinux label.
/proc/PID/attr/current

Equivalent of ps -eZ | ps -auxwwZ

**Comment by rhatdan** [comment]
Might also want to grab Capabilities.
grep Cap /proc/PID/status
The two we would care about are Eff and Bind.
Translating these into human understandable strings would also be huge.

**Comment by rhatdan** [comment]
Both of these could be done in future PRs

**Comment by vrothberg** [comment]
@rhatdan sounds great, I'll open two issues so we can track work.

**Comment by vrothberg** [comment]
> @rhatdan sounds great, I'll open two issues so we can track work.

You were faster :) Thanks!

**Pull request merged**